_posts/2018-09-05-Data-Continuity-Service-for-DynamoDB.md

```diff
@@ -0,0 +1,570 @@
+
+Let's start with creating a role in the source account which you will be assigning to the bucket in the source account. The source account is the account where the DynamoDB table is located.
+
+You can either create a bucket through the AWS Console or use the following snippet:
+
+```json
+{
+  "DynamoDbStreamToS3BucketRole": {
+        "Properties": {
+          "AssumeRolePolicyDocument": {
+            "Statement": [
+              {
+                "Action": [
+                  "sts:AssumeRole"
+                ],
+                "Effect": "Allow",
+                "Principal": {
+                  "Service": "s3.amazonaws.com"
+                }
+              }
+            ],
+            "Version": "2012-10-17"
+          },
+          "Path": "/",
+          "Policies": [
+            {
+              "PolicyDocument": {
+                "Statement": [
+                  {
+                    "Action": [
+                      "s3:GetObjectVersion",
+                      "s3:GetObjectVersionTagging",
+                      "s3:GetObjectVersionForReplication",
+                      "s3:GetObjectVersionAcl"
+                    ],
+                    "Effect": "Allow",
+                    "Resource": [
+                      "arn:aws:s3:::source-account-dynamodb-backup/*"
+                    ]
+                  },
+                  {
+                    "Action": [
+                      "s3:ListBucket",
+                      "s3:GetReplicationConfiguration"
+                    ],
+                    "Effect": "Allow",
+                    "Resource": [
+                      "arn:aws:s3:::source-account-dynamodb-backup"
+                    ]
+                  },
+                  {
+                    "Action": [
+                      "s3:ReplicateObject",
+                      "s3:ReplicateDelete",
+                      "s3:ReplicateTags",
+                      "s3:GetObjectVersionTagging",
+                      "s3:ObjectOwnerOverrideToBucketOwner"
+                    ],
+                    "Effect": "Allow",
+                    "Resource": [
+                      "arn:aws:s3:::destination-account-dynamodb-backup/*"
+                    ]
+                  }
+                ],
+                "Version": "2012-10-17"
+              },
+              "PolicyName": "DynamoDbStreamToS3BucketRole"
+            }
+          ]
+        },
+        "Type": "AWS::IAM::Role"
+      }  
+}
+```
+
+Next you need to create the source account bucket itself, keep in mind the bucket needs to be in the same region as where you did point your AWS Lambda function to (line 5). The bucket is configured with versioning (mandatory) and some lifecycle configuration policies to archive and cleanup these versioning. You can either copy these settings or define your own. Either way I would recommend you to set lifecycle policies as it will save you some extra bucks.
```

Comment:
Next you need to create the source account bucket itself. Keep in mind that the bucket needs to be in the same region as where you pointed your AWS Lambda function to (line 5). The bucket is configured with versioning (mandatory) and some lifecycle configuration policies to archive and cleanup versioned objects. You can either copy these settings or define your own. Either way I would recommend you to set lifecycle policies as it will save you some extra bucks.

Reply:
👍

Reply:
Thanks